packages/react-composites/src/composites/CallComposite/components/MediaGallery.tsx

```diff
@@ -117,3 +129,29 @@ export const useLocalVideoStartTrigger = (isLocalVideoAvailable: boolean, should
     }
   }, [shouldTransition, isButtonStatusSynced, isPreviewCameraOn, isLocalVideoAvailable, mediaGalleryHandlers]);
 };
+
+/**
+ * Hook to fetch new participant avatar and video gallery displayName information if it is present.
+ * @param onFetchAvatarPersonaData
+ */
+const useOnFetchAvatarPersonaData = (
+  remoteParticipants: VideoGalleryRemoteParticipant[],
+  onFetchAvatarPersonaData?: AvatarPersonaDataCallback
+): VideoGalleryRemoteParticipant[] => {
+  // Sort the remote participants so that they are always in order
+  remoteParticipants.sort((a, b) => (a.userId < b.userId ? 1 : -1));
+
+  const userIds = useMemo(() => {
+    return remoteParticipants.map((p) => p.userId);
+  }, [remoteParticipants]);
+
+  const avatarPersonaData = useDataProvider(userIds, onFetchAvatarPersonaData);
+  remoteParticipants.forEach((p, i) => {
+    if (avatarPersonaData[i]?.text === undefined) {
+      return;
+    }
+    p.displayName = avatarPersonaData[i]?.text;
```

Comment:
Similar to the side effects @prprabhu-ms mentioned above with sorting, this is mutating (changing) the displayName on the participant of the passed in remoteParticipants. This means the caller who uses `useOnFetchAvatarPersonaData` has just changed the participants display name for everyone using the remoteParticipants property.
Either this should be expected to mutate properties in place and not return anything (typically this is bad for usability, readability and debuggability of code)
Or (better) it should be returning a new set of remote participants with the augmentations in place (i.e. be a pure function with no side effects).

Reply:
okay will be sure to update this!